packages/edit-site/src/store/selectors.js

```diff
@@ -247,6 +248,61 @@ export function isSaveViewOpened( state ) {
 	return state.saveViewPanel;
 }
 
+/**
+ * Helper method to iterate through all blocks, recursing into inner blocks.
+ * Returns a flattened array of all blocks.
+ *
+ * @param {Array} blocks Blocks to flatten.
+ *
+ * @return {Array} Flattened object.
+ */
+function flattenBlocks( blocks ) {
+	const result = [];
+
+	const stack = [ ...blocks ];
+	while ( stack.length ) {
+		const { innerBlocks, ...block } = stack.shift();
+		// Place inner blocks at the beginning of the stack to preserve order.
+		stack.unshift( ...innerBlocks );
+		result.push( block );
+	}
+
+	return result;
+}
+
+function getFilteredTemplatePartBlocks( templateBlocks = [], templateParts ) {
```

Comment:
When referencing an empty array, we could declare an `EMPTY_ARRAY` constant and reference it here. IIRC, `memize()` is doing a strict comparison for each argument, and an empty array will be a different object every time, thus ignoring the previously cached value.

Reply:
Good catch! I've switched over to use `EMPTY_ARRAY`.